modules/bluefish/consumer/bluefish_consumer.cpp

```diff
@@ -75,7 +133,14 @@ struct bluefish_consumer : boost::noncopyable
 			
 	unsigned int										vid_fmt_;
 
-	std::array<blue_dma_buffer_ptr, 4>					reserved_frames_;	
+	std::array<blue_dma_buffer_ptr, 4>					all_frames_;	
+	std::queue<blue_dma_buffer_ptr>						reserved_frames_;
```

Comment:
I think it would be much simpler and less error prone to use `tbb::concurrent_queue` or `tbb::concurrent_bounded_queue` for both `reserved_frames_` and `live_frames_`. With no need for manual mutex handling.